### PR TITLE
Develop/keithley smu improvements

### PR DESCRIPTION
**Pull request by maij**
This PR has evolved dramatically, see below for the original intent.

This PR has now become basically a full driver rewrite. A lot of redundant code has been removed.

- [x] Remove a lot of redundant code
- [x] Replace numerous get/set cmds with a specialised SMUParameter
- [x] Add methods to get/set a voltage/current easily
- [x] Add methods to read from the SMU's event log
- [x] Add a wrapper for all functions checking for errors
   - ~~Incorporate timing checks to make sure it doesn't take too long~~
- [ ] Simplify/clarify docstrings  
- [ ] Clean up line spacing/indentation
---

Basic improvements to smu driver, parsing values to ints/float.

Added a feature to update the `source_level` and `sense_value` parameters' units when `source_mode` and `sense_mode` are changed, respectively.

I will postpone this until PR #52 is complete for EnumVisa

### CODE REVIEW

**Comment by nulinspiratie** [comment]
@maij ready to be reviewed?

File: qcodes/instrument_drivers/tektronix/Keithley_2450.py [hunk 1]
```diff
@@ -108,20 +123,24 @@ def __init__(self, name, address, **kwargs):
                            vals=Numbers(),
                            get_cmd=self._get_relative_offset,
                            set_cmd=self._set_relative_offset,
+                           get_parser=float,
+                           set_parser=float,
                            label='Relative offset value for a measurement.',
                            docstring='This specifies an internal offset that can be applied to measured data')
 
         self.add_parameter('relative_offset_state',
-                           val_mapping={'ON': 1, 'OFF': 0},
```
Comment (nulinspiratie) [inline qcodes/instrument_drivers/tektronix/Keithley_2450.py]: It's a tricky situation what would be the best here. I understand the reasoning for 0, 1.
Personally however, my preference goes out to boolean True / False, as it's the reason why the boolean was introduced.
What do you think? 
I see in this file that a lot of different terms are used ('ON'/'OFF', 'YES'/'NO', even 'True'/'False' in string format!)

**Comment by nulinspiratie** [comment]
@maij Is this ready to be reviewed again? Or do you want to tick off the remaining items first? Als othere's a conflict

**Comment by maij** [comment]
> @maij Is this ready to be reviewed again? Or do you want to tick off the remaining items first? Als othere's a conflict

Yeah, ready to be reviewed. I did some basic timing tests, and there isn't a noticeable slow down from adding error checks.
I can perform the neatening up steps later.

**Comment by nulinspiratie** [comment]
@maij I changed the formatting. This shouldn't affect the code, but it would be good if you could pull these changes and test it

**Pull request merged**